### PR TITLE
S28 3146: `GET /recordings` Add `sort` Param

### PR DESCRIPTION
**Pull request by lucas-phillips28**
### JIRA ticket(s)
- S28-3146

### Change description
- Change default sort order of `GET /recordings` endpoint to recording created at desc instead of case reference/version. 
- Exposed `sort` param on `GET /recordings` endpoint via `pageable` object to allow sorting in any order

> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Call `GET /recordings` endpoint- see response sorted by `created_at` value (most recent recordings first)
- Call `GET /recordings?sort=createdAt,asc` endpoint- see response sorted by `created_at` value (oldest recordings first)
- Call `GET /recordings` endpoint with other sort values to test it works for other sorting options
  - Could attempt to order by case reference (value should be `r.captureSession.booking.caseId.reference`)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/preapi/repositories/RecordingRepository.java [hunk 1]
```diff
@@ -98,17 +93,6 @@ Page<Recording> searchAllBy(
...
-        update #{#entityName} e
-        set e.deletedAt=CURRENT_TIMESTAMP
-        where e.captureSession=:captureSession
-        and e.deletedAt is null
-        """
-    )
-    @Modifying
-    @Transactional
-    void deleteAllByCaptureSession(CaptureSession captureSession);
-
```
Comment (jasonpaige) [inline src/main/java/uk/gov/hmcts/reform/preapi/repositories/RecordingRepository.java:111]: This doesn't seem relevant to the PR - is this just tidying up as you go along or has this been removed in error?

Comment (lucas-phillips28) [reply src/main/java/uk/gov/hmcts/reform/preapi/repositories/RecordingRepository.java:111]: This is just tidying up unused cod

File: src/main/java/uk/gov/hmcts/reform/preapi/controllers/RecordingController.java [hunk 2]
```diff
@@ -127,7 +134,9 @@ public ResponseEntity<RecordingDTO> getRecordingById(
     @PreAuthorize("hasAnyRole('ROLE_SUPER_USER', 'ROLE_LEVEL_1', 'ROLE_LEVEL_2', 'ROLE_LEVEL_3', 'ROLE_LEVEL_4')")
     public HttpEntity<PagedModel<EntityModel<RecordingDTO>>> searchRecordings(
         @Parameter(hidden = true) @ModelAttribute SearchRecordings params,
-        @Parameter(hidden = true) Pageable pageable,
+        @SortDefault.SortDefaults(
+            @SortDefault(sort = "createdAt", direction = Sort.Direction.DESC)
+        ) @Parameter(hidden = true) Pageable pageable,
```
Comment (jasonpaige) [inline src/main/java/uk/gov/hmcts/reform/preapi/controllers/RecordingController.java:139]: nice - v tidy 👍

**Review by jasonpaige — APPROVED** [review]

**Comment by NazeeKadiu** [comment]
Testing has been completed and passing.

**Review by NazeeKadiu — APPROVED** [review]
Testing completed and passed
